doc/ChangeLog

```diff
@@ -1,4 +1,108 @@
 ===============================================================
+Tag name: ctsm5.2.018
+Originator(s): mvdebolskiy
+Date: Wed 31 Jul 2024 04:38:36 PM MDT
```

Comment:
@mvdebolskiy this is a start at the ChangeLog please review and lets make any important changes needed.